Mathlib/RingTheory/FinitePresentation.lean

```diff
@@ -415,11 +416,9 @@ theorem id : FinitePresentation (RingHom.id A) :=
 variable {A}
 
 theorem comp_surjective {f : A →+* B} {g : B →+* C} (hf : f.FinitePresentation) (hg : Surjective g)
-    (hker : g.ker.FG) : (g.comp f).FinitePresentation :=
-  letI := f.toAlgebra
-  letI := (g.comp f).toAlgebra
-  letI : Algebra.FinitePresentation A B := hf
-  Algebra.FinitePresentation.of_surjective
+    (hker : g.ker.FG) : (g.comp f).FinitePresentation := by
+  algebraize f (g.comp f)
```

Comment:
I'm not super excited about it taking a sequence of terms, with no separators. Seems like an invitation for confusing parsing. Would it be a problem to *either* take a single terms, or to take a list `[f, g, (g.comp f)]`?

Reply:
I don't know if there is a standard on this.

We could definitely do `algebraize [f, g]` following the example of [`Lean.Parser.Tactic.rwRuleSeq`](https://leanprover-community.github.io/mathlib4_docs/Init/Tactics.html#Lean.Parser.Tactic.rwRuleSeq)

`simp` even does this directly in-place:

```lean
syntax (name := simp) "simp" (config)? (discharger)? (&" only")?
  (" [" withoutPosition((simpStar <|> simpErase <|> simpLemma),*,?) "]")? (location)? : tactic
```

It seems reasonable to do it the same way in that case.

Reply:
This seems like a good idea.

Reply:
I agree, I like the list suggestion! Will try to incorporate it soon.

Reply:
Implemented this :)